### ISSUE TITLE
My game crashed

### ISSUE THREAD

**Issue by deleted-user-1** [OpenRA/OpenRA#18218]
## System Information

- Operating System:  Linux/Ubuntu 28.04
- OpenRA Version: 7a78c378512c3092d407090f807a3ab2efe48702
- OpenRA Mod: RA
- Source: self-compiled
- For self-compiled or third-party packages:  Mono 6.8.0.123

## Exception log

```
OpenRA engine version {DEV_VERSION}
Red Alert mod version {DEV_VERSION}
on map 208f2eee47675e702f08517322eb2a69dd0f221b (Sidestep by SoScared).
Date: 2020-06-01 17:04:03Z
Operating System: Linux (Unix 4.15.0.101)
Runtime Version: Mono 6.8.0.123 (tarball Tue May 12 15:11:57 UTC 2020) CLR 4.0.30319.42000
Exception of type `System.IndexOutOfRangeException`: Index was outside the bounds of the array.
  at OpenRA.CellLayer`1[T].get_Item (OpenRA.CPos cell) [0x0000d] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Mods.Common.Traits.LegacyBridgeLayer.ConvertBridgeToActor (OpenRA.World w, OpenRA.CPos cell) [0x00125] in <c6a9cbed3ef94b08a206734da225ef88>:0 
  at OpenRA.Mods.Common.Traits.LegacyBridgeLayer.WorldLoaded (OpenRA.World w, OpenRA.Graphics.WorldRenderer wr) [0x00101] in <c6a9cbed3ef94b08a206734da225ef88>:0 
  at OpenRA.World.LoadComplete (OpenRA.Graphics.WorldRenderer wr) [0x000c4] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.StartGame (System.String mapUID, OpenRA.WorldType type) [0x000ce] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Network.UnitOrders.ProcessOrder (OpenRA.Network.OrderManager orderManager, OpenRA.World world, System.Int32 clientId, OpenRA.Order order) [0x00724] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Network.OrderManager.TickImmediate () [0x000c9] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Sync+<>c__DisplayClass13_0.<RunUnsynced>b__0 () [0x00000] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Sync.RunUnsynced[T] (System.Boolean checkSyncHash, OpenRA.World world, System.Func`1[TResult] fn) [0x00003] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Sync.RunUnsynced (System.Boolean checkSyncHash, OpenRA.World world, System.Action fn) [0x0000d] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x000fb] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.LogicTick () [0x0003e] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.Loop () [0x000f1] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.Run () [0x0003c] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00044] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
```

## Additional information

This crashed happened when starting the game, ironically it was displaying the "Building bridges" status text when the terminal showed the exception. I had to kill the process. ~~I can't reproduce it so perhaps it was a memory error or something different where my system failed at?~~

**Comment by deleted-user-1** [comment]
It just happened again, so probably something is wrong.

**Comment by deleted-user-1** [comment]
Maybe it is my system that is broken, I see some more errors lately. Last one showed 
```
AL lib: ReleaseALC: 1 device not closed
Segmentation fault (core dumped)
```
at the end of the exception that i posted above in the terminal after I waited a bit and didn't kill the process. Feel free to close if you can't see a relation to OpenRA code since I have no idea how to reproduce it.

**Comment by abcdefg30** [comment]
I can reproduce the crash when starting Sidestep, certainly not limited to your device.

**Comment by deleted-user-1** [comment]
Yeah, i just got another one in TS while the game was running in the background (map `They all float`):

```
Exception of type `System.IndexOutOfRangeException`: Index was outside the bounds of the array.
  at OpenRA.CellLayer`1[T].get_Item (OpenRA.CPos cell) [0x0000d] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Map.DistanceAboveTerrain (OpenRA.WPos pos) [0x00035] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Mods.Common.Activities.Move+MovePart.UpdateCenterLocation (OpenRA.Actor self, OpenRA.Mods.Common.Traits.Mobile mobile) [0x000bd] in <c6a9cbed3ef94b08a206734da225ef88>:0 
  at OpenRA.Mods.Common.Activities.Move+MovePart.Tick (OpenRA.Actor self) [0x0002d] in <c6a9cbed3ef94b08a206734da225ef88>:0 
  at OpenRA.Activities.Activity.TickOuter (OpenRA.Actor self) [0x00084] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Traits.ActivityUtils.RunActivity (OpenRA.Actor self, OpenRA.Activities.Activity act) [0x00015] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Activities.Activity.TickChild (OpenRA.Actor self) [0x00008] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Activities.Activity.TickOuter (OpenRA.Actor self) [0x0007c] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Traits.ActivityUtils.RunActivity (OpenRA.Actor self, OpenRA.Activities.Activity act) [0x00015] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Activities.Activity.TickChild (OpenRA.Actor self) [0x00008] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Mods.Common.Activities.AttackMoveActivity.Tick (OpenRA.Actor self) [0x000b2] in <c6a9cbed3ef94b08a206734da225ef88>:0 
  at OpenRA.Activities.Activity.TickOuter (OpenRA.Actor self) [0x000b5] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Traits.ActivityUtils.RunActivity (OpenRA.Actor self, OpenRA.Activities.Activity act) [0x00015] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Actor.Tick () [0x0000f] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.World.Tick () [0x00123] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001bc] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.LogicTick () [0x0003e] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.Loop () [0x000f1] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.Run () [0x0003c] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00044] in <8e18d02ea2ce41e3bff62c02626d0afe>:0 
```

**Comment by abcdefg30** [comment]
That crash is likely fixed by #18209 and a bit different to the first.
